client/valor/schemas/symbolic/types.py

```diff
@@ -0,0 +1,1372 @@
+        else:
+            return True
+
+    @classmethod
+    def decode_value(cls, value: Any):
+        """Decode object from JSON compatible dictionary."""
+        return cls(value=value)
+
+    def encode_value(self) -> Any:
+        """Encode object to JSON compatible dictionary."""
+        return self.get_value()
+
+    @classmethod
+    def from_dict(cls, value: dict):
+        """Decode a JSON-compatible dictionary into an instance of the variable."""
+        if set(value.keys()) != {"type", "value"}:
+            raise KeyError
+        elif value["type"] != cls.__name__.lower():
+            raise TypeError
+        return cls.decode_value(value["value"])
+
+    def to_dict(self) -> dict:
+        """Encode variable to a JSON-compatible dictionary."""
+        if isinstance(self._value, Symbol):
+            return self._value.to_dict()
+        else:
+            return {
+                "type": type(self).__name__.lower(),
+                "value": self.encode_value(),
+            }
+
+    @property
+    def is_symbolic(self) -> bool:
+        """Returns whether variable is symbolic."""
+        return isinstance(self._value, Symbol)
+
+    @property
+    def is_value(self) -> bool:
+        """Returns whether variable contains a value."""
+        return not isinstance(self._value, Symbol)
+
+    def get_value(self) -> Any:
+        """Retrieve value, if it exists."""
+        if isinstance(self._value, Symbol):
+            raise TypeError(
+                f"{type(self).__name__} is symbolic and does not contain a value."
+            )
+        return self._value
+
+    def get_symbol(self) -> Symbol:
+        """Retrieve symbol, if it exists."""
+        if not isinstance(self._value, Symbol):
+            raise TypeError(f"{type(self).__name__} is a valued object.")
+        return self._value
+
+    def __eq__(self, value: Any) -> Union["Bool", Eq]:  # type: ignore - overriding __eq__
+        raise AttributeError(
+            f"'{type(self).__name__}' object has no attribute '__eq__'"
+        )
+
+    def __ne__(self, value: Any) -> Union["Bool", Ne]:  # type: ignore - overriding __ne__
+        raise AttributeError(
+            f"'{type(self).__name__}' object has no attribute '__ne__'"
+        )
+
+    def __gt__(self, value: Any) -> Union["Bool", Gt]:
+        raise AttributeError(
+            f"'{type(self).__name__}' object has no attribute '__gt__'"
+        )
+
+    def __ge__(self, value: Any) -> Union["Bool", Ge]:
+        raise AttributeError(
+            f"'{type(self).__name__}' object has no attribute '__ge__'"
+        )
+
+    def __lt__(self, value: Any) -> Union["Bool", Lt]:
+        raise AttributeError(
+            f"'{type(self).__name__}' object has no attribute '__lt__'"
+        )
+
+    def __le__(self, value: Any) -> Union["Bool", Le]:
+        raise AttributeError(
+            f"'{type(self).__name__}' object has no attribute '__le__'"
+        )
+
+
+class Bool(Variable):
+    """
```

Comment:
nit: the docstrings in this file are pretty inconsistent. some are just one liners, some have examples, and some are full docstrings. ideally, all public methods will have full docstrings

Reply:
Ive updated most of them!